### PR TITLE
feat: invalidate rfox unstaking queries on claim/unstake

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Does what it says on the box:

- invalidates rfox unstaking queries after claim/unstake Tx is succesful
- while at it, removes `refetchOnMount` from the claim query, which I assume the original refetch intent was as a bandaid for the fact we didn't invalidate. Having this behavior makes the UX quite messy with many many refetches and skeletons.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/7239

Low to none

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Claims list is properly refetched after a claim/unstake Tx is success

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

https://jam.dev/c/a68dffd8-f24f-4278-8321-1efe3d99e5c0

### CODE REVIEW

File: src/pages/RFOX/components/Unstake/Unstake.tsx [hunk 1]
```diff
@@ -70,12 +72,28 @@ export const UnstakeRoutes: React.FC<UnstakeRouteProps> = ({ headerComponent })
...
+    stakingAssetAccountAddress: confirmedQuote
+      ? getAddress(fromAccountId(confirmedQuote.stakingAssetAccountId).account)
+      : undefined,
+  })
+
   const handleTxConfirmed = useCallback(async () => {
     await queryClient.invalidateQueries({ queryKey: userStakingBalanceOfCryptoBaseUnitQueryKey })
     await queryClient.invalidateQueries({ queryKey: newContractBalanceOfCryptoBaseUnitQueryKey })
+    await queryClient.invalidateQueries({ queryKey: unstakingRequestCountQueryKey })
+    await queryClient.invalidateQueries({ queryKey: unstakingRequestQueryKey })
```
Comment (woodenfurniture) [inline src/pages/RFOX/components/Unstake/Unstake.tsx:91]: 🧠

File: src/pages/RFOX/components/Claim/Claim.tsx [hunk 2]
```diff
@@ -44,10 +49,28 @@ export const Claim: React.FC<ClaimRouteProps> = ({ headerComponent, setStepIndex
...
+
+  const { queryKey: unstakingRequestQueryKey } = useGetUnstakingRequestsQuery({
+    stakingAssetAccountAddress: confirmedQuote
+      ? getAddress(fromAccountId(confirmedQuote.stakingAssetAccountId).account)
+      : undefined,
+  })
+
+  const handleTxConfirmed = useCallback(async () => {
+    await queryClient.invalidateQueries({ queryKey: unstakingRequestCountQueryKey })
+    await queryClient.invalidateQueries({ queryKey: unstakingRequestQueryKey })
```
Comment (woodenfurniture) [inline src/pages/RFOX/components/Claim/Claim.tsx:71]: preferably blocking: If the intent here is only to invalidate the corresponding queries matching the query key, it could be lighter on the app to grab the query key only above without invoking the query via `useGetUnstakingRequestCountQuery` and `useGetUnstakingRequestsQuery`, otherwise we can have a case where we invoke the query, and then invalidate it immediately

An example of the pattern to expose query key getters is in `src/pages/RFOX/hooks/useEpochHistoryQuery.ts` where a `getEpochHistoryQueryKey` is a separate function to the hook itself, allowing you to invalidate a query without invoking it in the case none matches the key you generated, reducing overhead in that case.

File: src/pages/RFOX/components/Unstake/Unstake.tsx [hunk 3]
```diff
@@ -70,12 +72,28 @@ export const UnstakeRoutes: React.FC<UnstakeRouteProps> = ({ headerComponent })
...
+    stakingAssetAccountAddress: confirmedQuote
+      ? getAddress(fromAccountId(confirmedQuote.stakingAssetAccountId).account)
+      : undefined,
+  })
+
+  const { queryKey: unstakingRequestQueryKey } = useGetUnstakingRequestsQuery({
+    stakingAssetAccountAddress: confirmedQuote
+      ? getAddress(fromAccountId(confirmedQuote.stakingAssetAccountId).account)
+      : undefined,
+  })
```
Comment (woodenfurniture) [inline src/pages/RFOX/components/Unstake/Unstake.tsx:85]: preferably blocking: likewise here, per https://github.com/shapeshift/web/pull/7240/files#r1655620212

**Review by woodenfurniture — CHANGES_REQUESTED** [review]
Generally LGTM with 2 comments preferably blocking to potentially reduce redundant fetching

File: src/pages/RFOX/components/Claim/Claim.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/pages/RFOX/components/Claim/Claim.tsx:71]: Generally would agree with that, though those queries are a bit particular ones:

What we do with those wagmi wrappers is effectively reconstructing wagmi hooks (e.g `useReadContract`) in a type-safe way (i.e supporting `skipToken`), using the same core utils as them and having matching `queryKey`s to be aligned with their implementation, meaning when wagmi bumps their react-query version to v5, we should be able to ditch our hand-rolled hooks, and simply consume the wagmi variant.

image

we could make the queryKey-building a function instead of an expression, however there is one main issue with that:

If (hopefully wen, doubt wagmi is going to be stuck on v4 forever) wagmi bumps to v5 and we ditch our hand-rolled hooks, then we're good-to-go with the current approach as [this is also how wagmi does it](https://wagmi.sh/react/guides/migrate-from-v1-to-v2#removed-watch-property) - however, if we were to implement some query-buliding function instead, then we wouldn't be able to remove our hand-rolled hooks.

tl;dr everytime we see some hook that's effectively our hand-rolled wagmi hook remade from source, just assume that the API is similar to wagmi's and that we don't have access to internals in a way that would let us e.g build the queryKey in isolation

Comment (woodenfurniture) [reply src/pages/RFOX/components/Claim/Claim.tsx:71]: Ah sure that makes sense in terms of formulating a drop-in replacement so we can upgrade to wagmi v5 semlessly, though the question remains:
If the query for the given key has not fired, does us invalidating it cause it to fetch then invalidate then fetch again, effectively 2 fetches?
I'm going to stamp so we can get this moving as it seems more of a theoretical concern

**Review by woodenfurniture — APPROVED** [review]
